climpred/relative_entropy.py

```diff
@@ -0,0 +1,343 @@
+import warnings
+from collections import OrderedDict
+
+import matplotlib.pyplot as plt
+import numpy as np
+import xarray as xr
+from eofs.xarray import Eof
+
+
+def _relative_entropy_formula(sigma_b, sigma_x, mu_x, mu_b, neofs):
+    """
+    Computes the relative entropy formula given in Branstator and Teng, (2010).
+
+    References:
+        - Branstator, Grant, and Haiyan Teng. “Two Limits of Initial-Value
+            Decadal Predictability in a CGCM.” Journal of Climate 23, no. 23
+            (August 27, 2010): 6292–6311. https://doi.org/10/bwq92h.
+        - Kleeman, Richard. “Measuring Dynamical Prediction Utility Using
+            Relative Entropy.” Journal of the Atmospheric Sciences 59, no. 13
+            (July 1, 2002): 2057–72. https://doi.org/10/fqwxpk.
+
+    Args:
+        sigma_b (xr.DataArray): covariance matrix of baseline distribution
+        sigma_x (xr.DataArray): covariance matrix of forecast distribution
+        mu_b (xr.DataArray): mean state vector of the baseline distribution
+        mu_x (xr.DataArray): mean state vector of the forecast distribution
+        neofs (int): number of EOFs used
+
+    Returns:
+        R (float): relative entropy
+        dispersion (float): dispersion component
+        signal (float): signal component
+    """
+    fac = 0.5
+    dispersion = fac * (np.log(np.linalg.det(sigma_b) / np.linalg.det(sigma_x))
+                        + np.trace(sigma_x / sigma_b) - neofs)
+
+    # https://stackoverflow.com/questions/7160162/left-matrix-division-and-numpy-solve
+    # (A.T\B)*A
+    x, resid, rank, s = np.linalg.lstsq(
+        sigma_b, mu_x - mu_b)  # sigma_b \ (mu_x - mu_b)
+    signal = fac * np.matmul((mu_x.values - mu_b.values), x)
+    R = dispersion + signal
+    return R, dispersion, signal
+
+
+def _bootstrap_dim(control, lead_years, time_dim='initialization', dim='member', dim_label=list(np.arange(10))):
```

Comment:
This `dim_label` keyword doesn't make sense to me. Could you just create it in the function and default to the number of members in `ds`? Why is 10 the default? For instance, DPLE has 40 members and other ensembles may have other numbers than 10.

Reply:
I choose 10 as my setup has 10 members. I want to have it flexible in the args because I might need to increase the number. see notebook bottom.